desktop/src/client/core/aad/proxy-network-client.ts

```diff
@@ -0,0 +1,55 @@
+import type { INetworkModule, NetworkRequestOptions, NetworkResponse } from "@azure/msal-node";
+import * as HttpsProxyAgent from "https-proxy-agent";
+import fetch from "node-fetch";
+
+/**
+ * Placeholder for msal-node's network module which uses node-fetch to support
+ * HTTP proxy configurations with authorization
+ *
+ * @see https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/6527#issuecomment-2077953882
+ */
+export class ProxyNetworkClient implements INetworkModule {
+    private proxyAgent: HttpsProxyAgent;
+    constructor(proxyUrl: string) {
+        this.proxyAgent = new HttpsProxyAgent(proxyUrl);
+    }
+
+    sendGetRequestAsync<T>(url: string, options?: NetworkRequestOptions): Promise<NetworkResponse<T>> {
+        return this.sendRequestAsync(url, "GET", options);
+    }
+    sendPostRequestAsync<T>(url: string, options?: NetworkRequestOptions): Promise<NetworkResponse<T>> {
+        return this.sendRequestAsync(url, "POST", options);
+    }
+
+    private async sendRequestAsync<T>(
+        url: string,
+        method: "GET" | "POST",
```

Comment:
What about other HTTP verbs? (PUT/PATCH/DELETE/etc)

Reply:
We don't need to handle other verbs because the Entra handshake only involves `GET` and `POST`. This network client is only being used for authentication.